src/Adapter/AbstractObjectModelRepository.php

```diff
@@ -66,15 +72,30 @@ protected function assertObjectModelExists(int $id, string $objectTableName, str
      * @param int $id
      * @param string $objectModelClass
      * @param string $exceptionClass
+     * @param ShopId|null $shopId
      *
      * @return ObjectModel
      *
      * @throws CoreException
      */
-    protected function getObjectModel(int $id, string $objectModelClass, string $exceptionClass): ObjectModel
+    protected function getObjectModel(int $id, string $objectModelClass, string $exceptionClass, ?ShopId $shopId = null): ObjectModel
```

Comment:
IMO this calls for a `NullShopId` class and this method issuing a deprecated soft error when passed null.

Reply:
Now that I think of it, I can't find a scenario where you wouldn't need the shop id, if you pass null then you're just leaving it up to Context

Reply:
> On second thought I wonder if this parameter should not be mandatory because not all the ObjectModel handle multishop (actually the majority of them don't..)

Every ObjectModel is supposed to be able to be multistore, but enabling or disabling this feature is done for each case. Just like any field can be translatable but it's enabled or not for each field.

So it's OK for me to consider that the standard usecase is "handle ObjectModel for different Shop context" and that "when option is disabled" is a specific case.

Reply:
Mmm, actually the ObjectModel does handle multishop internally, but not all the Entities use this feature, that's what I meant when I said that not all types of ObjectModel are multishop

And the thing is I believe there are more entities that don't use it than the opposite, that's why I think it's preferable that the abstract repository has dedicated base methods for both use cases This way you can use the appropriate one depending on the type of ObjectModel you are using